lib/ddtrace/context.rb

```diff
@@ -15,18 +15,45 @@ module Datadog
   # This data structure is thread-safe.
   class Context
     # Initialize a new thread-safe \Context.
-    def initialize
+    def initialize(options = {})
       @mutex = Mutex.new
-      reset
+      reset(options)
     end
 
-    def reset
+    def reset(options = {})
       @trace = []
-      @sampled = false
+      @parent_trace_id = options.fetch(:trace_id, nil)
+      @parent_span_id = options.fetch(:span_id, nil)
+      @sampled = options.fetch(:sampled, false)
+      @sampling_priority = options.fetch(:sampling_priority, nil)
       @finished_spans = 0
       @current_span = nil
     end
 
+    def trace_id
+      @mutex.synchronize do
+        return @parent_trace_id
```

Comment:
If you don't mind changing this, let's get rid of the `return` just to be more idiomatic.

Reply:
:+1: